### ISSUE TITLE
Preserve path in git repository

### ISSUE THREAD

**Issue by abeMedia** [stormcat24/protodep#22]
Hi @stormcat24, first of all thanks for writing such a handy tool!

I have a use case where it would be great to preserve to directory structure within the cloned repo in my `proto_outdir` (e.g. `github.com/googleapis/googleapis/google/type` is written to `${proto_outdir}/google/type`). 

Basically I'm importing some dependencies which rely on `github.com/googleapis/googleapis/google/type` and use the import path `google/type/${name}.proto`, but I don't require any other files from that repo so I'm trying to avoid setting the dependency target to `github.com/googleapis/googleapis` and having a load of unnecessary files in my repository.

I would like to suggest adding a global option called `preserve_path` or similar to facilitate this.
If that's something you feel is not in the scope of protodep, an alternative solution would be allowing to specify a path on a dependency i.e.:

```toml
[[dependencies]]
  target = "github.com/googleapis/googleapis/google/type"
  branch = "master"
  path = "google/type"
```

I personally prefer the first method of using a global option, however the second method of specifying a path is definitely more dynamic and could help with other use-cases such as name collisions. What do you think?

**Comment by stormcat24** [comment]
@abeMedia Sounds good. I think the second method is the best for protodep.

**Comment by abeMedia** [comment]
Fixed in #23
